File: WeWorkInTheDark/submissions/time_limit_exceeded/bruteforce.py
#! /usr/bin/env python3
# import time

# start_time = time.time()
input_str = input()
input_list = input_str.split(" ")
x = int(input_list[0])
y = int(input_list[1])
area = [[0 for _ in range(y)] for _ in range(x)]
n = int(input_list[2])

for _ in range(n):
    guards = input().split(" ")
    guards_x = int(guards[0])
    guards_y = int(guards[1])
    guards_amount = int(guards[2])
    area[guards_x][guards_y] += guards_amount
    #print(f"{guards_x} {guards_y} updated: {area[guards_x][guards_y]}")

m = int(input().strip())

for _ in range(m):
    query = input().split(" ")
    command = query[0]

    if command == "e":
        ezio_x = int(query[1])
        ezio_y = int(query[2])
        ezio_range = int(query[3])
        x_lower_bound = max(0, ezio_x - ezio_range)
        y_lower_bound = max(0, ezio_y - ezio_range)
        x_upper_bound = min(x-1, ezio_x + ezio_range)
        y_upper_bound = min(y-1, ezio_y + ezio_range)
        #bounds = f"Bounds: {x_lower_bound} {y_lower_bound} {x_upper_bound} {y_upper_bound}"
        #print(bounds)
        amount = 0
        for i in range(x_lower_bound, x_upper_bound+1):
            for j in range(y_lower_bound, y_upper_bound+1):
                amount += area[i][j]
        print(amount)

    elif command == "r":
        r_x = int(query[1])
        r_y = int(query[2])
        r_amount = int(query[3])
        area[r_x][r_y] += r_amount

    elif command == "k":
        k_x = int(query[1])
        k_y = int(query[2])
        area[k_x][k_y] = 0
""" end_time = time.time()
elapsed_time = end_time - start_time
print(f"Elapsed time: {elapsed_time * 1000} milliseconds") """

File: WeWorkInTheDark/submissions/accepted/bruteforce.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class bruteforce {
    public static void main(String[] args) throws IOException {
        long startTime = System.currentTimeMillis();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String inputStr = br.readLine();
        String[] inputList = inputStr.split(" ");
        int x = Integer.parseInt(inputList[0]);
        int y = Integer.parseInt(inputList[1]);
        int[][] area = new int[x][y];
        int n = Integer.parseInt(inputList[2]);

        for (int i = 0; i < n; i++) {
            String[] guards = br.readLine().split(" ");
            int guards_x = Integer.parseInt(guards[0]);
            int guards_y = Integer.parseInt(guards[1]);
            int guards_amount = Integer.parseInt(guards[2]);
            area[guards_x][guards_y] += guards_amount;
            // System.out.println(guards_x + " " + guards_y + " updated: " + area[guards_x][guards_y]);
        }

        int m = Integer.parseInt(br.readLine().trim());

        for (int i = 0; i < m; i++) {
            String[] query = br.readLine().split(" ");
            String command = query[0];

            if (command.equals("e")) {
                int ezio_x = Integer.parseInt(query[1]);
                int ezio_y = Integer.parseInt(query[2]);
                int ezio_range = Integer.parseInt(query[3]);
                int x_lower_bound = Math.max(0, ezio_x - ezio_range);
                int y_lower_bound = Math.max(0, ezio_y - ezio_range);
                int x_upper_bound = Math.min(x - 1, ezio_x + ezio_range);
                int y_upper_bound = Math.min(y - 1, ezio_y + ezio_range);
                // String bounds = "Bounds: " + x_lower_bound + " " + y_lower_bound + " " + x_upper_bound + " " + y_upper_bound;
                // System.out.println(bounds);
                int amount = 0;
                for (int row = x_lower_bound; row <= x_upper_bound; row++) {
                    for (int col = y_lower_bound; col <= y_upper_bound; col++) {
                        amount += area[row][col];
                    }
                }
                System.out.println(amount);
            } else if (command.equals("r")) {
                int r_x = Integer.parseInt(query[1]);
                int r_y = Integer.parseInt(query[2]);
                int r_amount = Integer.parseInt(query[3]);
                area[r_x][r_y] += r_amount;
            } else if (command.equals("k")) {
                int k_x = Integer.parseInt(query[1]);
                int k_y = Integer.parseInt(query[2]);
                area[k_x][k_y] = 0;
            }
        }

        long endTime = System.currentTimeMillis();
        long elapsed_time = endTime - startTime;
        System.out.println("Elapsed time: " + elapsed_time + " milliseconds");
    }
}
